server/src/main/java/org/elasticsearch/action/search/MultiSearchRequest.java

```diff
@@ -297,7 +297,7 @@ public static void writeSearchRequestParams(SearchRequest request, XContentBuild
         if (request.indices() != null) {
             xContentBuilder.field("index", request.indices());
         }
-        if (request.indicesOptions() != null && request.indicesOptions() != SearchRequest.DEFAULT_INDICES_OPTIONS) {
+        if (request.indicesOptions() != null && request.indicesOptions().equals(SearchRequest.DEFAULT_INDICES_OPTIONS) == false) {
```

Comment:
I think we no longer need the condition `request.indicesOptions() != null` as `ndicesOptions()` is always non-null.

Reply:
Thanks, I pushed https://github.com/elastic/elasticsearch/pull/71688/commits/7f53b16402c1f99c1f6ee231234122759c073dd5